### ISSUE TITLE
Gameplay addition n°3: Decoy Turret

### ISSUE THREAD

**Issue by MartinCHTITI** [l0th3r/TD-Template#9]
We would like to add a turret that does not attack enemies but insead becomes the units' main objective as long as it's active.

**Comment by l0th3r** [comment]
Yes of course, I would like you to specify the behaviour with a specification of every new mechanics added, like doing a list of every new or modified elements.

**Comment by l0th3r** [comment]
Thank you for your reply.

I understand the wanted feature.
To explain how I see it:

![decoy-turret_explanations](https://user-images.githubusercontent.com/58365407/170814146-2abe8744-f1cf-40ff-8115-b25ec705f293.png)

Units cannot go backward about the path, for example if a entity is on `p2` it cannot go back to `p1`.
But, When entering a platform, an **entity performs multiple checks**:
- Is my behaviour defined as `AttackOnSight` or `ReachEnd`  ([what ?](https://github.com/l0th3r/TD-Template/wiki/Entity#td_entity_minion))  and am I at the end of my path ? => then I attack all buildings near the platform I am in. (The objective normally or all adjacent buildings on the last platforms.)
- If not, is my behaviour defined as `AttackOnSight` => then I attack all building near the platform I am in.
- Else I either move to the next platform or I stop if no platform left.

So no, an unit cannot go backward but ! When entering a a new platform those checks are performed. It means that when a turret is added, the unit will only know when reaching a new platform. For example imagine the turret is not placed, if a unit is in `p2`, it will see that the turret is not there. So it moves at `p3` and then performs it's checks. So if a turret has been placed during the movement between the two platforms well it doesn't matter, the entity will attack the new turret if it is told to do so.

So, if a unit is moving from `p3` to `p4` no, during the movement it will not see if a decoy turret is added. But when reaching another platform it will check any way if a new turret is near. So when entering `p1`, `p2` or `p3`, a unit will be catched by the decoy turret. Not before entering `p1` nor after starting the movement off  `p3`.

I hope that I answered your questions about units and their behaviour.
This functionality can be added before June 5 it will depend on the problems that can occur or if I did not understand what you wanted.
